delta-lake/common/src/main/scala/com/nvidia/spark/rapids/delta/GpuRapidsRepartitionByFilePath.scala

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2020-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.delta
+
+import com.nvidia.spark.rapids._
+import com.nvidia.spark.rapids.delta.RapidsRepartitionByFilePath.FILE_PATH_FIELD
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.rdd.{PartitionCoalescer, PartitionGroup, RDD}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, UnaryNode}
+import org.apache.spark.sql.catalyst.plans.physical.{HashPartitioning, Partitioning}
+import org.apache.spark.sql.execution.{SparkPlan, SparkStrategy, UnaryExecNode}
+import org.apache.spark.sql.types.{LongType, StringType, StructField}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+object RapidsRepartitionByFilePath {
+  val ROW_ID_COL = "_metadata_row_id"
+  val ROW_ID_FIELD = StructField(ROW_ID_COL, LongType, nullable = false)
+  val FILE_PATH_COL = "_metadata_file_path"
+  val FILE_PATH_FIELD = StructField(FILE_PATH_COL, StringType, nullable = false)
+}
+
+case class RapidsRepartitionByFilePath(optNumPartition: Option[Int],
```

Comment:
Comments on these classes and what they're doing would help.  Somewhere there needs to be an overview of how these custom operators plugin into the low shuffle merge algorithm.  I haven't seen any comments explaining why we're partitioning on the file path as part of the algorithm, for example.